doc/developer-guides/memmgt-hld.rst

```diff
@@ -0,0 +1,248 @@
+.. _memmgt-hld:
+
+Memory Management High-Level Design
+###################################
+
+This document describes memory management for the ACRN hypervisor.
+
+Overview
+********
+
+In the ACRN hypervisor system, there are few different memory spaces to
+consider.  From the hypervisor's point of view there are:
+
+-  Host Physical Address (HPA): the native physical address space, and
+-  Host Virtual Address (HVA): the native virtual address space based on
+   a MMU. A page table is used to do the translation between HPA and HVA
+   spaces.
+
+And from the Guest OS running on a hypervisor there are:
+
+-  Guest Physical Address (GPA): the guest physical address space from a
+   virtual machine.  GPA to HPA transition is usually based on a
+   MMU-like hardware module (EPT in X86), and associated with a page
+   table
+-  Guest Virtual Address (GVA): the guest virtual address space from a
+   virtual machine based on a vMMU
+
+.. figure:: images/mem-image2.png
+   :align: center
+   :width: 900px
+   :name: mem-overview
+
+   ACRN Memory Mapping Overview
+
+:numref:`mem-overview` provides an overview of the ACRN system memory
+mapping, showing:
+
+-  GVA to GPA mapping based on vMMU on a VCPU in a VM
+-  GPA to HPA mapping based on EPT for a VM in the hypervisor
+-  HVA to HPA mapping based on MMU in the hypervisor
+
+This document illustrates the memory management infrastructure for the
+ACRN hypervisor and how it handles the different memory space views
+inside the hypervisor and from a VM:
+
+-  How ACRN hypervisor manages host memory (HPA/HVA)
+-  How ACRN hypervisor manages SOS guest memory (HPA/GPA)
+-  How ACRN hypervisor & SOS DM manage UOS guest memory (HPA/GPA)
+
+Hypervisor Memory Management
+****************************
+
+The ACRN hypervisor is the primary owner to manage system
+memory. Typically the boot firmware (e.g., EFI) passes the platform physical
+memory layout - E820 table to the hypervisor. The ACRN hypervisor does its memory
+management based on this table.
+
+Physical Memory Layout - E820
+=============================
+
+The BIOS/bootloader passes the E820 table through a multiboot protocol.
```

Comment:
Can you also update this to `boot firmware (e.g., EFI)`? (sorry I had missed it earlier).